### PR TITLE
adding evaluation code

### PR DESCRIPTION
**Pull request by 51N84D**
Short program to infer model on a folder of images

### CODE REVIEW

File: eval.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+
+if __name__ == "__main__":
+    # -----------------------------
+    # -----  Parse arguments  -----
+    # -----------------------------
+
+    args = parsed_args()
+    output_dir = Path(args.output_dir)
+    output_dir.mkdir(exist_ok=True)
```
Comment (vict0rsch) [inline eval.py:65]: you could add `parents=True` so that we could specify a hierarchy without checking all parents exist

File: eval.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        new_size = args.new_size
+
+    # ------------------------
+    # ----- Define model -----
+    # ------------------------
+
+    trainer = Trainer(opts)
+    trainer.setup()
+    trainer.resume()
+    model = trainer.G
```
Comment (vict0rsch) [inline eval.py:88]: add `model.eval()` to skip gradients

File: eval.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    ]
+
+    device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+
+    # ----------------------------
+    # -----  Iterate images  -----
+    # ----------------------------
+
+    images = [args.path_to_images / Path(i) for i in os.listdir(args.path_to_images)]
+    for img_path in images:
```
Comment (vict0rsch) [inline eval.py:106]: this would make the code a little more complicated but:

what about adding a `batch_size` arg so that if there are many images we could process them in batches

File: eval.py (hunk 1 above)
Comment (51N84D) [reply eval.py:65]: True

File: eval.py (hunk 3 above)
Comment (sashavor) [reply eval.py:107]: Good idea!

Comment (51N84D) [reply eval.py:107]: Can we merge this for now, and add that as a feature request? The new code does an os.walk on the image directory and progressively generates the images from each folder (while preserving the folder structure).

It might be a bit more involved to do batching here
